ethereum/core/src/main/java/org/hyperledger/besu/ethereum/privacy/UnrestrictedPrivacyController.java

```diff
@@ -0,0 +1,298 @@
+    throw new PrivacyConfigurationNotSupportedException(
+        "Method not supported for UnrestrictedPrivacy");
+  }
+
+  @Override
+  public long determineEeaNonce(
+      final String privateFrom,
+      final String[] privateFor,
+      final Address address,
+      final String enclavePublicKey) {
+
+    final String privacyGroupId = createPrivacyGroupId(privateFrom, privateFor);
+    return determineBesuNonce(address, privacyGroupId, enclavePublicKey);
+  }
+
+  private String createPrivacyGroupId(final String privateFrom, final String[] privateFor) {
+    final Bytes32 privacyGroupId =
+        PrivacyGroupUtil.calculateEeaPrivacyGroupId(
+            Bytes.fromBase64String(privateFrom),
+            Arrays.stream(privateFor).map(Bytes::fromBase64String).collect(Collectors.toList()));
+
+    return privacyGroupId.toBase64String();
+  }
+
+  @Override
+  public long determineBesuNonce(
+      final Address sender, final String privacyGroupId, final String enclavePublicKey) {
+    return privateNonceProvider.getNonce(
+        sender, Bytes32.wrap(Bytes.fromBase64String(privacyGroupId)));
+  }
+
+  @Override
+  public Optional<TransactionProcessingResult> simulatePrivateTransaction(
+      final String privacyGroupId,
+      final String enclavePublicKey,
+      final CallParameter callParams,
+      final long blockNumber) {
+    return privateTransactionSimulator.process(privacyGroupId, callParams, blockNumber);
+  }
+
+  @Override
+  public Optional<Bytes> getContractCode(
+      final String privacyGroupId,
+      final Address contractAddress,
+      final Hash blockHash,
+      final String enclavePublicKey) {
+
+    return privateWorldStateReader.getContractCode(privacyGroupId, blockHash, contractAddress);
+  }
+
+  @Override
+  public PrivateTransactionSimulator getTransactionSimulator() {
+    return privateTransactionSimulator;
+  }
+
+  @Override
+  public Optional<Hash> getStateRootByBlockNumber(
+      final String privacyGroupId, final String enclavePublicKey, final long blockNumber) {
+    return blockchain
+        .getBlockByNumber(blockNumber)
+        .map(
+            block ->
+                privateStateRootResolver.resolveLastStateRoot(
+                    Bytes32.wrap(Bytes.fromBase64String(privacyGroupId)), block.getHash()));
+  }
+
+  @Override
+  public Optional<String> buildAndSendAddPayload(
+      final PrivateTransaction privateTransaction,
+      final Bytes32 privacyGroupId,
+      final String enclaveKey) {
+    throw new PrivacyConfigurationNotSupportedException(
+        "Method not supported for UnrestrictedPrivacy");
+  }
+
+  @Override
+  public Optional<PrivacyGroup> findOffChainPrivacyGroupByGroupId(
+      final String toBase64String, final String enclaveKey) {
+    throw new PrivacyConfigurationNotSupportedException(
+        "Method not supported for UnrestrictedPrivacy");
+  }
+
+  @Override
+  public Optional<PrivacyGroup> findPrivacyGroupByGroupId(
```

Comment:
This method is only here because it's used by `PrivDebugGetStateRoot` to validate the existence of something. I'm not sure it's actually needed in `PrivDebugGetStateRoot` but figured it must be there for a reason. 

Either way, it's only used to validate that someone has called/set up the distribution part of the group which isn't needed for unrestricted privacy.

Reply:
Added to https://github.com/hyperledger/besu/issues/2370